Fix the generation of tokens with specific scopes and disallow token with empty scope

MVrachev opened the pull request:
Fix a bug where the generation of a token with a scope containing `read:tasks` or `delete:targets` lead to an error.
Also, disallow the generation of a token where the `scopes` field is an empty list.

I added regression tests for both changes.

Close #185 #187 

Signed-off-by: Martin Vrachev <mvrachev@vmware.com>

MVrachev commented on tests/unit/api/test_token.py:
```diff
@@ -77,12 +77,47 @@ def test_post(self, monkeypatch, test_client):
             "username": "admin",
             "password": "secret",
-            "scope": "write:targets",
+            "scope": (
+                "read:bootstrap read:settings read:tasks read:token "
+                "write:targets delete:targets"
+            ),
```
Here I added all allowed scopes to make sure we can create a token containing each of them and such bugs as #187 won't happen for other scopes unnoticed.

kairoaraujo commented on tests/unit/api/test_token.py:
```diff
@@ -77,12 +77,47 @@ def test_post(self, monkeypatch, test_client):
         assert response.status_code == status.HTTP_200_OK
         assert "access_token" in response.json()
 
+    def test_post_new(self, test_client, token_headers):
```
nit: Add both tests after `def test_post_with_missing_scope`

kairoaraujo reviewed (changes requested):
I would add two more tests

```python
    def test_post_new_with_scopes_with_empty_string(self, test_client, token_headers):
        url = "/api/v1/token/new/"
        payload = {
            "scopes": [''],
            "expires": 1,
        }

        response = test_client.post(url, headers=token_headers, json=payload)
        assert response.status_code == status.HTTP_422_UNPROCESSABLE_ENTITY
        assert "unexpected value" in response.json()["detail"][0]["msg"]
        
    def test_post_new_with_scopes_with_invalid_scope(self, test_client, token_headers):
        url = "/api/v1/token/new/"
        payload = {
            "scopes": ["invalid"],
            "expires": 1,
        }

        response = test_client.post(url, headers=token_headers, json=payload)
        assert response.status_code == status.HTTP_422_UNPROCESSABLE_ENTITY
        assert "unexpected value" in response.json()["detail"][0]["msg"]

MVrachev commented:
I added the two tests you suggested.
I decided to move the `post_new` token tests next to the rest of the `post_new` tests.
I am not sure what is the connection between the tests I added and the `test_post_with_missing_scope` test. 
Do you think it's better to move the tests there or leave them where I suggest right now?

kairoaraujo commented:
> I added the two tests you suggested.
> I decided to move the `post_new` token tests next to the rest of the `post_new` tests.
> I am not sure what is the connection between the tests I added and the `test_post_with_missing_scope` test.
> Do you think it's better to move the tests there or leave them where I suggest right now?

I like the way you did it. All `test_pos_*` and all `test_post_new_*`

kairoaraujo reviewed (approved):
LGTM! Ready to merge.
